Web.R.tex

```diff
@@ -115,37 +120,45 @@ \chapter{Background}\label{ch:background}
 We introduce discussion about the decentralized Web by defining a key concept that allows us to better explain the difference between the standard Web and its distributed\footnote{In this work, we will use the terms ``decentralized'' and ``distributed'' as synonyms. Although their meaning is slightly different, that difference is not relevant in this work and we choose to not consider it.} counterpart: failure models.
 
 \section{Failure models}
+We consider a distributed system composed of \emph{processes} which communicate by sending messages through communication \emph{channels}.
+The set of processes includes servers (e.g., web, naming, storage), clients (e.g., browsers) and other entities such as routers.
+%todo(leandro) - I think this definition mimics what you previously wrote in the end of the "reference model"
 
-The two failure models that we define here are the \emph{stop} (or \emph{stopping}) failure model and the Byzantine failure model.
+In this work, we consider two failure models: \emph{Fail-stop} (or \emph{stopping}) and \emph{Byzantine}.
 
-%todo - needed?
-In the following definitions, we consider \emph{processes} any entity that has an active function and interaction in a system. If we apply this concept to the Web, we consider to be processes servers, clients and communication channels (including routers), to name the most important ones.
+In the Fail-stop failure model, a process may fail by interrupting all activities. This is known as \emph{halting} or \emph{crashing} in some cases. Once a process has halted, it remains in that state. Other processes are able to detect this failure using some detection technique. In the case of the Web, this is tipically done through \emph{timeouts}: if we expect a communication that does not occur within a time limit, we consider the process to have failed.
+We include in the failure model the behaviour of communication channels: they may drop, reorder and delay messages, but may not create or alter them.
+Furthermore, if a message is sent infinitely often, it is also received infinitely often.
+In practice, that means processes can always communicate (i.e., find a network route to each other), except during temporary network disruptions.
```

Comment:
I think this part on communication channels is way too complex. I don't go into that much detail during the various analysis (I often don't even mention them) so I think this much detail is not needed.

Reply:
Makes sense.
How about the following?

We include in the failure model the behavior of communication channels. 
In the stopping model, messages may be dropped or delayed, but not created or modified.
We assume processes can always communicate (i.e., find a network route to each other), except during temporary network disruptions.